logFilter.js

```diff
@@ -12,12 +12,22 @@ function filterLogsByEnv(logs, env) {
 
 function filterLogsOnDate(logs, dateFilter) {
   const filteredLogs = []
-  const parsedDateFilter = new Date(dateFilter)
-
-  logs.forEach((log) => {
-    const timestampDate = new Date(log.timestamp)
-
-    if (parsedDateFilter.toDateString() === timestampDate.toDateString()) {
+  let parsedDateFilter = new Date(dateFilter)
+  parsedDateFilter = new Date(
+    parsedDateFilter.getFullYear(),
+    parsedDateFilter.getUTCDate(),
+    parsedDateFilter.getUTCDay()
+  )
+
+  logs.forEach(log => {
+    let timestampDate = new Date(log.timestamp)
+    timestampDate = new Date(
+      timestampDate.getFullYear(),
+      timestampDate.getUTCDate(),
+      timestampDate.getUTCDay()
+    )
```

Comment:
Here you are creating a date object from a unix timestamp. Unix timestamps are always in UTC and date objects are always in UTC so you have a date object that is the time the log happened in UTC. 

The new date you are creating is actually not going to be correct. Date() expects year, month, day. You are giving it 2019, 28, 4 since you are using getUTCDate and getUTCDay. That actually gives you `2021-05-04T04:00:00.000Z` because the 28 for month makes everything wonky.

Reply:
Yea I noticed that the date was way off and I was thinking if this program was used for something else it would not work. I kept trying all kind of things and I know it was simple but I was doing alot of over thinking on this one. I just am not getting the date function. I am going to trying to rework it again so I can get it right.

Reply:
Definitely a case of looking easier than it is. But I have no doubt you will crack it, just gotta keep at it.